lib/gui/pages/main/controllers/flash.js

```diff
@@ -26,10 +26,50 @@ module.exports = function(
   FlashErrorModalService,
   ErrorService,
   OSNotificationService,
-  OSWindowProgressService
+  OSWindowProgressService,
+  DriveInfoModalService
 ) {
 
   /**
+   * @summary Flash image to a drive, with a confirmation modal on system drives
+   * @function
+   * @public
+   *
+   * @param {String} image - image path
+   * @param {Object} drive - drive
+   *
+   * @example
+   * FlashController.flashImageToDrive('rpi.img', {
+   *   device: '/dev/disk2',
+   *   description: 'Foo',
+   *   size: 99999,
+   *   mountpoint: '/mnt/foo',
+   *   system: false
+   * });
+   */
+  this.flashImageToDriveModal = (image, drive) => {
```

Comment:
This variable name implies that `this.flashImageToDriveModal` represents the modal itself, and not a function that opens something else (the `DriveInfoModal` component). Maybe we can rename to something like `askFlashConfirmation` or something like that?